src/modules/hunting/etcd.py

```diff
@@ -81,66 +50,57 @@ def db_keys_write_access(self):
 
+
 # Passive Hunter
 @handler.subscribe(OpenPortEvent, predicate=lambda p: p.port == 2379)
 class EtcdRemoteAccess(Hunter):
     """Etcd Remote Access
     Checks for remote availability of etcd, version, read access, write access
     """
-    # TODO:
-    # Check the etcd hunter on a remote cluster! (currently everything was checked only at 127.0.0.1:2379)
+
     def __init__(self, event):
         self.event = event
         self.version_evidence = ''
         self.keys_evidence = ''
+        self.protocol = 'https'
 
     def db_keys_disclosure(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to read etcd keys remotely")
-        r_secure = "https://{host}:{port}/v2/keys".format(host=self.event.host, port=2379)
-        r_not_secure = "http://{host}:{port}/v2/keys".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_secure, r_not_secure)
-        if res:
-            self.keys_evidence = res.content
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to read etcd keys remotely")
+        r = requests.get(
+            "{protocol}://{host}:{port}/v2/keys".format(protocol=self.protocol, host=self.event.host, port=2379),
+            verify=False)
+        self.keys_evidence = r.content if r.status_code == 200 and r.content != '' else False
+        return self.version_evidence
 
     def version_disclosure(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to check etcd version remotely")
-        r_secure = "https://{host}:{port}/version".format(host=self.event.host, port=2379)
-        r_not_secure = "http://{host}:{port}/version".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_secure, r_not_secure)
-        if res:
-            self.version_evidence = res.content
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to check etcd version remotely")
+        r = requests.get(
+            "{protocol}://{host}:{port}/version".format(protocol=self.protocol, host=self.event.host, port=2379),
+            verify=False)
+        self.version_evidence = r.content if r.status_code == 200 and r.content != '' else False
+        return self.version_evidence
 
     def unauthorized_access(self):
-        logging.debug(self.event.host)
-        logging.debug("Passive hunter is attempting to access etcd without authorization")
-        r_not_secure = "http://{host}:{port}/version".format(host=self.event.host, port=2379)
-        res = helperFuncDo2Requests(r_not_secure, r_not_secure)  # We don't have to do 2 requests this time
-        if res:
-            return True
-        return False
+        logging.debug(self.event.host + " Passive hunter is attempting to access etcd without authorization")
```

Comment:
The code is fine but the comment and function name are misleading. Lots of people find authentication and authorization confusing so let me try to explain. 
* Here we're making an insecure (http) request, passing no identity information. 
* You can also make a secure (https) request without passing identity information. This encrypts the communication. 
* If you pass identity information (like a bearer token, or a user name & password) the other end might "authenticate" you i.e. check your identity by verifying that the password is correct for the username. There's not a lot of point passing identity information if you're not also encrypting the communication, otherwise anyone could just pick up your identity and pretend to be you, so realistically an authenticated connection will also be encypted (https), but not all https traffic is authenticated. 
* If you are authenticated, the other knows who you are, and might then also make you go through an "authorization" process: "now that I know who you are, I'm going to check whether you have the authority to make this request". 

So in this case it's really just trying to access etcd unencrypted (aka insecure), over http.